Content.Server/_Citadel/Shipyard/Systems/ShipyardSystem.cs

```diff
@@ -0,0 +1,177 @@
+using System;
+using Content.Server.Shuttles.Systems;
+using Content.Server.Shuttles.Components;
+using Content.Server.Station.Components;
+using Content.Server.Cargo.Systems;
+using Content.Server.Station.Systems;
+using Content.Shared.MobState.Components;
+using Content.Shared.GameTicking;
+using Robust.Server.GameObjects;
+using Robust.Server.Maps;
+using Robust.Shared.Map;
+using Robust.Shared.GameObjects;
+using static Content.Server.Power.Pow3r.PowerState;
+
+namespace Content.Server.Shipyard.Systems
+{
+
+    public sealed partial class ShipyardSystem : EntitySystem
+    {
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        [Dependency] private readonly PricingSystem _pricing = default!;
+        [Dependency] private readonly ShuttleSystem _shuttle = default!;
+        [Dependency] private readonly StationSystem _station = default!;
+        [Dependency] private readonly CargoSystem _cargo = default!;
+        [Dependency] private readonly MapLoaderSystem _map = default!;
+
+        public MapId? ShipyardMap { get; private set; }
+        private float _shuttleIndex;
+        private const float ShuttleSpawnBuffer = 1f;
+        private ISawmill _sawmill = default!;
+
+        public override void Initialize()
+        {
+            _sawmill = Logger.GetSawmill("shipyard");
+            SubscribeLocalEvent<BecomesStationComponent, ComponentStartup>(OnShipyardStartup);
+            SubscribeLocalEvent<RoundRestartCleanupEvent>(OnRoundRestart);
+        }
+
+        private void OnShipyardStartup(EntityUid uid, BecomesStationComponent component, ComponentStartup args)
+        {
+            SetupShipyard();
+        }
+
+        private void OnRoundRestart(RoundRestartCleanupEvent ev)
+        {
+            CleanupShipyard();
+        }
+
+        /// <summary>
+        /// Adds a ship to the shipyard, calculates its price, and attempts to ftl-dock it to the given station
+        /// </summary>
+        /// <param name="Station ID"></param>
+        /// <param name="Shuttle Path"></param>
```

Comment:
```suggestion
        /// <param name="stationUid">The ID of the station to do a purchase for.</param>
        /// <param name="shuttlePath">The path to the shuttle map file to load. This should be a grid!</param>
```
You'll additionally need to fix all other XML docs.